### ISSUE TITLE
Contrast themes – Modal dialogs need outline or opaque backdrop

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#8833]
In Windows high contrast mode / forced-colors mode, our dialog component lacks a backdrop, thus making it hard to understand where the component is (and the fact it’s modal).

Screenshot (left: WHCM, right: default):

![Image](https://user-images.githubusercontent.com/877585/178252282-7a56a85d-e04a-4cc1-b11a-cb41b0107d01.png)

Figma link: https://www.figma.com/file/O81E48wieOZ7e6AivvksWn/Contrast-themes%3A-Wagtail-audit?node-id=6766%3A4147

----

Edit: we are clear! `outline: 20px solid transparent`.

~We’re not clear on what the ideal solution for this is, so it’ll need a bit of initial R&D / discussion before we settle on a specific pattern.~

Options seem to be:

- Adding a thick outline, as suggested by @SaptakS as part of @wagtail/accessibility discussions: https://www.smashingmagazine.com/2022/03/windows-high-contrast-colors-mode-css-custom-properties/#using-custom-properties-with-forced-colors-mode
- Making the backdrop fully opaque, suggested in [web-a11y Slack](https://web-a11y.slack.com/archives/C02D66WBDTQ/p1657203369545429)
- Keeping the backdrop partly transparent with `forced-color-adjust: none`
